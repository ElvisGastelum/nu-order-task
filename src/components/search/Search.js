import React from 'react';

import SearchItemsList from 'components/search-items-list';

import {
  SearchContainer,
  SearchElements,
  SearchBox,
  SearchBoxLogo,
  SearchInputWrapper,
  SearchInput,
  SearchTitle,
} from 'components/search/styled';

import githubLogo from './img/github-logo.png';

/**
 * Render a search component with suggestions
 *
 * @param {{ newSearchResults: (value: string) => void, searchResults: [], index: number, setArrowKeys: (keyCode: number, index?: number) => void, handleFocus: (e: React.FocusEvent<HTMLInputElement>) => void, handleBlur: (e: React.FocusEvent<HTMLInputElement>) => void }} props used for the component
 * @return {JSX.Element} the JSX.Element
 */
export default function Search({
  newSearchResults,
  searchResults,
  index,
  setArrowKeys,
  backgroundActive,
}) {
  const handleText = (e) => newSearchResults(e.target.value);
  const handleArrowKeys = (e) => setArrowKeys(e.keyCode);
  const handleFocus = (e) => backgroundActive(false);
  const handleBlur = (e) => backgroundActive(true);

  return (
    <SearchContainer>
      <SearchElements>
        <SearchBox>
          <SearchInputWrapper>
            <SearchBoxLogo src={githubLogo} alt="github-logo" />
            <SearchTitle>Search React Issues</SearchTitle>
            <SearchInput
              type="text"
              placeholder="Search"
              onChange={handleText}
              onKeyDown={handleArrowKeys}
              onFocus={handleFocus}
              onBlur={handleBlur}
            />
            <SearchItemsList
              items={searchResults}
              index={index}
              setArrowKeys={setArrowKeys}
            />
          </SearchInputWrapper>
        </SearchBox>
      </SearchElements>
    </SearchContainer>
  );
}
