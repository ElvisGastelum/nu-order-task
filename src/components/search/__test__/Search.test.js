import React from 'react';

import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { act } from 'react-dom/test-utils';

import Search from 'components/search';
import data from './mockData.json';
import { searchIssues } from 'hooks/use-suggestions/searchIssues';

Enzyme.configure({
  adapter: new Adapter(),
});

describe('Test <Search /> component', () => {
  test('should be render', async () => {
    var searchResults = [];
    const newSearchResults = (value) => {
      searchResults = searchIssues(value, data);
    };
    var index = 0;
    const setArrowKeys = (e) => {};

    const wrapper = mount(
      <Search
        searchResults={searchResults}
        newSearchResults={newSearchResults}
        index={index}
        setArrowKeys={setArrowKeys}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should be regenerate html when pass new props', async () => {
    var searchResults = [];
    const newSearchResults = (value) => {
      searchResults = searchIssues(value, data);
    };
    var index = 0;
    const setArrowKeys = (e) => {};

    const wrapper = mount(
      <Search
        searchResults={searchResults}
        newSearchResults={newSearchResults}
        index={index}
        setArrowKeys={setArrowKeys}
      />
    );

    const initialHTML = wrapper.html();

    act(() => {
      wrapper.find('input').simulate('change', { target: { value: 'good' } });
    });

    act(() => {
      wrapper.setProps({
        searchResults,
      });
    });

    expect(wrapper.html()).not.toBe(initialHTML);

    expect(wrapper).toMatchSnapshot();
  });

  test('should be have a length 5 when search "react" label in the input', async () => {
    var searchResults = [];
    const newSearchResults = (value) => {
      searchResults = searchIssues(value, data);
    };
    var index = 0;
    const setArrowKeys = (e) => {};

    const wrapper = mount(
      <Search
        searchResults={searchResults}
        newSearchResults={newSearchResults}
        index={index}
        setArrowKeys={setArrowKeys}
      />
    );

    const initialHTML = wrapper.html();

    act(() => {
      wrapper.find('input').simulate('change', { target: { value: 'react' } });
    });

    act(() => {
      wrapper.setProps({
        searchResults,
      });
    });

    expect(wrapper.html()).not.toBe(initialHTML);

    expect(wrapper.find('ul').children().length).toBe(5);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be have a length 3 when search "good first issue" label in the input', async () => {
    var searchResults = [];
    const newSearchResults = (value) => {
      searchResults = searchIssues(value, data);
    };
    var index = 0;
    const setArrowKeys = (e) => {};

    const wrapper = mount(
      <Search
        searchResults={searchResults}
        newSearchResults={newSearchResults}
        index={index}
        setArrowKeys={setArrowKeys}
      />
    );

    const initialHTML = wrapper.html();

    act(() => {
      wrapper.find('input').simulate('change', { target: { value: 'good first issue' } });
    });

    act(() => {
      wrapper.setProps({
        searchResults,
      });
    });

    expect(wrapper.html()).not.toBe(initialHTML);

    expect(wrapper.find('ul').children().length).toBe(3);
    expect(wrapper).toMatchSnapshot();
  });
});
