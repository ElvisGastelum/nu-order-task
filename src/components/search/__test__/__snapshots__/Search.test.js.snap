// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Search /> component should be have a length 3 when search "good first issue" label in the input 1`] = `ReactWrapper {}`;

exports[`Test <Search /> component should be have a length 5 when search "react" label in the input 1`] = `ReactWrapper {}`;

exports[`Test <Search /> component should be regenerate html when pass new props 1`] = `ReactWrapper {}`;

exports[`Test <Search /> component should be render 1`] = `ReactWrapper {}`;
